boutdata/restart.py

```diff
@@ -1047,7 +1048,10 @@ def change_grid(
     # Read information from a restart file
     with DataFile(file_list[0]) as f:
         for var in copy_vars:
-            copy_data[var] = f[var]
+            try:
+                copy_data[var] = f[var]
+            except:
+                print("Failed to copy variable '{}'".format(var))
```

Comment:
```suggestion
            if var not in f:
                print(f"{var} not present - skipping")
                continue
            copy_data[var] = f[var]
```

That has the advantage of only skipping if the variable isn't there, but not if the file is or implementation is broken.

Reply:
Do you have cases where variables are missing? Old BOUT++ simulations? Or is it failing for other reasons?